spec/models/customization_template_spec.rb

```diff
@@ -0,0 +1,30 @@
+describe CustomizationTemplate do
+  context "unique name validation" do
+    let(:pxe_image_type)                { FactoryGirl.create(:pxe_image_type) }
+    let(:customization_template)        { FactoryGirl.create(:customization_template, :pxe_image_type => pxe_image_type) }
+    let(:system_customization_template) { described_class.create!(:system => true, :name => "nick") }
+
+    it "doesn't allow same name in same region" do
+      FactoryGirl.create(:customization_template, :name => "fred")
+      expect { FactoryGirl.create(:customization_template, :name => "fred") }.to raise_error(ActiveRecord::RecordInvalid)
+    end
+
+    it "allows different names with the same pxe_image_type in same region" do
+      FactoryGirl.create(:customization_template, :name => "fred", :pxe_image_type => pxe_image_type)
+      FactoryGirl.create(:customization_template, :name => "nick", :pxe_image_type => pxe_image_type)
+      expect(described_class.count).to eq(2)
+    end
+
+    it "allow same name, nil pxe_image_types in different regions (for system seeding)" do
+      ctid = ApplicationRecord.id_in_region(system_customization_template.id, ApplicationRecord.my_region_number + 1)
+      expect { described_class.create!(:system => true, :name => "nick", :id => ctid) }.to change(described_class, :count).by(1)
+    end
+
+    it "allows same name, different pxe_image_types in different regions" do
+      pid =  ApplicationRecord.id_in_region(pxe_image_type.id, MiqRegion.my_region_number + 1)
+      ctid = ApplicationRecord.id_in_region(customization_template.id, MiqRegion.my_region_number + 1)
+      pt2 =  FactoryGirl.create(:pxe_image_type, :id => pid)
+      expect { FactoryGirl.create(:customization_template, :pxe_image_type => pt2, :id => ctid) }.to change(described_class, :count).by(1)
```

Comment:
A name needs to be specified in this `create` call and in the `let` for `customization_template`. name is a sequence in the factory so the name of this record won't be the same as `customization_template.name`

Reply:
Interesting, let me see if the test fails if I check that correctly.  Good 👀